pkg/checkpoint/secret.go

```diff
@@ -50,14 +53,71 @@ func (c *checkpointer) checkpointSecret(namespace, podName, secretName string, u
 
 	// TODO(aaron): No need to store if already exists
 	for f, d := range secret.Data {
-		if err := writeAndAtomicRename(filepath.Join(basePath, f), d, uid, gid, 0600); err != nil {
+		if err := writeAndAtomicRename(filepath.Join(basePath, f), d, uid, gid, defaultSecretMode); err != nil {
 			return "", fmt.Errorf("failed to write secret %s: %v", secret.Name, err)
 		}
 	}
 
 	return basePath, nil
 }
 
+// checkpointSecretProjection will locally store secret data from volumes with ProjectedVolumeSources.
+// The path to the secret data becomes: checkpointSecretPath/namespace/podname/volumename/secret.file
+// Where each "secret.file" is a path from the SecretProjection.KeyToPath field.
+func (c *checkpointer) checkpointSecretProjection(namespace, podName, volumeName string, secretProjection *v1.SecretProjection, uid, gid int) error {
```

Comment:
nit: This seems to duplicate a fair amount of behavior with checkpointConfigMapProjection (on a quick scan differences are the checkpoint path and retrieving the object from api). Might be able to merge those behaviors with something like a `checkpointProjection` that just takes one of the object types.